### PR TITLE
Connect search by post api to client

### PR DESCRIPTION
**Pull request by DwinaTech**
Connect search by post api to client

### CODE REVIEW

File: frontend/src/Components/Organisation/index.css [hunk 1]
```diff
@@ -31,4 +31,11 @@
   font-weight: bold;
   color: #1abcd4;
   outline: none !important;
-}
\ No newline at end of file
+}
+
+.distanse{
```
Comment (kkarimi) [inline frontend/src/Components/Organisation/index.css]: not formatted

File: frontend/src/Components/Organisation/index.js [hunk 2]
```diff
@@ -76,7 +82,37 @@ class Organisations extends Component {
...
+            .map(branchs => {
+              const { distance } = branchs;
+              const orgs = branchs.data;
+              return orgsData.push({ distance, ...orgs })
+            })
+          this.setState({ organisations: orgsData })
+        })
+        this.setState({ isLoading: false })
+      } else {
+        this.setState({ isLoading: false })
```
Comment (kkarimi) [inline frontend/src/Components/Organisation/index.js]: Ideally one function should not setState multiple times during one journey, you should set multiple values in one place, e.g. line 111+112, etc (Do check all the setStates here!)

File: frontend/src/Components/Organisation/index.js [hunk 3]
```diff
@@ -105,9 +141,18 @@ class Organisations extends Component {
...
+    if (!this.state.sort) {
+      return helpers.sortArrObj
+    }
+    return (a, b) => parseFloat(a.distance) - parseFloat(b.distance)
+  }
+
   render() {
     const { editIdx, category, postCode, borough, day, organisations } = this.state;
-
+    if (this.state.isLoading || this.filterData.length <= 0) {
```
Comment (kkarimi) [inline frontend/src/Components/Organisation/index.js]: Can length be less than 0?

File: frontend/src/actions/postData.js [hunk 4]
```diff
@@ -13,3 +13,11 @@ export function editOrganisation(data) {
     axios.patch(`${api}/service/organisation/edit`, data).then(res => res);
   return saveOrganisation
 }
+
+export function getBranchesFilteredByPost(data) {
```
Comment (kkarimi) [inline frontend/src/actions/postData.js]: getBranchesFilteredByPost -> `getBranchesFilteredByPostCode`

File: frontend/src/actions/postData.js [hunk 5]
```diff
@@ -13,3 +13,11 @@ export function editOrganisation(data) {
...
+  const sendInfo = () =>
+    axios.post(`${api}/service/postcode`, data)
```
Comment (kkarimi) [inline frontend/src/actions/postData.js]: Guess what I would say here :)

File: frontend/src/Components/Organisation/index.css (hunk 1 above)
Comment (DwinaTech) [reply frontend/src/Components/Organisation/index.css]: I do not understand, what do you mean not formatted?

File: frontend/src/Components/Organisation/index.js (hunk 3 above)
Comment (DwinaTech) [reply frontend/src/Components/Organisation/index.js]: Good question, no.

File: frontend/src/actions/postData.js (hunk 5 above)
Comment (DwinaTech) [reply frontend/src/actions/postData.js]: try and catch

File: frontend/src/actions/postData.js (hunk 4 above)
Comment (DwinaTech) [reply frontend/src/actions/postData.js]: Done

File: frontend/src/Components/Organisation/index.js (hunk 2 above)
Comment (DwinaTech) [reply frontend/src/Components/Organisation/index.js]: Done

File: frontend/src/Components/Organisation/index.js (hunk 3 above)
Comment (DwinaTech) [reply frontend/src/Components/Organisation/index.js]: corrected to better logic

File: frontend/src/actions/postData.js (hunk 5 above)
Comment (DwinaTech) [reply frontend/src/actions/postData.js]: Done

File: frontend/src/Components/Organisation/index.css (hunk 1 above)
Comment (DwinaTech) [reply frontend/src/Components/Organisation/index.css]: alright

Comment (DwinaTech) [reply frontend/src/Components/Organisation/index.css]: @kkarimi Done

**Pull request merged**